### PR TITLE
reactive design/code mode toggle

### PR DESCRIPTION
**Pull request by Bjvanminnen**
designtoggle

Replace our code/design button with a reactive element. We also show a dropdown when in design mode. Right now that will never change, but eventually it will show a list of our screen elements. We could arguably have that dropdown be a separate child component.

I also moved our drag/drop configuration code into designMode.js

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
Just realized I had the text wrong for the button. Fixed. (i.e. If I'm in design mode, the button should say Code not Design)

File: apps/src/applab/DesignToggleRow.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+
+  getInitialState: function () {
+    return {
+      mode: Mode.CODE
+    };
+  },
+
+  handleModeToggle: function () {
+    var newMode;
```
Comment (davidsbailey) [inline apps/src/applab/DesignToggleRow.jsx]: remove newMode

File: apps/src/applab/designMode.js [hunk 2]
```diff
@@ -16,7 +19,7 @@ var currentlyEditedElement = null;
  * @param event
  */
 designMode.onDivApplabClick = function (event) {
-  if ($('#designModeButton').is(':visible') ||
+  if (Applab.isInDesignMode() ||
```
Comment (Bjvanminnen) [inline apps/src/applab/designMode.js]: Should be !Applab.isInDesignMode()

File: apps/src/applab/DesignToggleRow.jsx (hunk 1 above)
Comment (davidsbailey) [inline apps/src/applab/DesignToggleRow.jsx]: oops, you can leave it this way, or you could put the setState call inline. either way is fine.

File: apps/src/applab/designMode.js [hunk 3]
```diff
@@ -369,3 +367,56 @@ function makeDraggable (jq) {
...
+designMode.configureDesignToggleRow = function () {
+  var designToggleRow = document.getElementById('designToggleRow');
+  if (!designToggleRow) {
+    return;
+  }
+
+  // TODO (brent) - still need logic to generate list of screens, and rerender
+  // DesignToggleRow on changes
+  React.render(
+    React.createElement(DesignToggleRow, {
```
Comment (davidsbailey) [inline apps/src/applab/designMode.js]: The design button (or perhaps this whole component) needs to be hidden behind user.isAdmin . This is how we are hiding unlaunched features from field testers for now.

File: apps/src/applab/designMode.js [hunk 4]
```diff
@@ -369,3 +367,56 @@ function makeDraggable (jq) {
     }
   });
 }
+
+designMode.configureDragAnDrop = function () {
```
Comment (davidsbailey) [inline apps/src/applab/designMode.js]: configureDragAnDrop --> configureDragAndDrop (assuming unintentional misspelling)

File: apps/src/applab/designMode.js (hunk 3 above)
Comment (davidsbailey) [inline apps/src/applab/designMode.js]: ah, never mind -- I see this is still hidden behind the same data.hasDesignMode.

**Comment by davidsbailey** [comment]
LGTM after addressing one minor comment

**Pull request merged**